Source/Core/Core/PowerPC/MMU.cpp

```diff
@@ -166,10 +166,10 @@ BatTable dbat_table;
 
 static void GenerateDSIException(u32 effective_address, bool write);
 
-template <XCheckTLBFlag flag, typename T, bool never_translate = false>
+template <XCheckTLBFlag flag, typename T, TranslateMode mode = HostTranslate>
 static T ReadFromHardware(u32 em_address)
 {
-  if (!never_translate && MSR.DR)
+  if (((mode == HostTranslate) && MSR.DR) || mode == AlwaysTranslate)
```

Comment:
One cannot assume in general that translation lookup is set-up correctly when MSR[DR]=0.